commcare-cloud-bootstrap/commcare_cloud_bootstrap.py

```diff
@@ -124,27 +174,83 @@ def ask_aws_for_instances(env, aws_config, count):
             '--tag-specifications', 'ResourceType=instance,Tags=[{Key=env,Value=' + env + '}]',
         ]
         aws_response = subprocess.check_output(cmd_parts)
-        with open('ask_aws_for_instances.json', 'w') as f:
+        with open(cache_file, 'w') as f:
             f.write(aws_response)
-
     aws_response = json.loads(aws_response)
     return {instance['InstanceId'] for instance in aws_response["Instances"]}
 
 
 def print_describe_instances(describe_instances):
     for reservation in describe_instances['Reservations']:
         for instance in reservation['Instances']:
-            print("{InstanceId}\t{InstanceType}\t{ImageId}\t{State[Name]}\t{PublicIpAddress}\t{PublicDnsName}".format(**instance),
+            print("{InstanceId}\t{InstanceType}\t{ImageId}\t{State[Name]}\t{PublicIpAddress}\t{PrivateIpAddress}".format(**instance),
                   file=sys.stderr)
 
 
-def poll_for_aws_state(env, instance_ids):
+def raw_describe_instances(env):
     cmd_parts = [
         'aws', 'ec2', 'describe-instances', '--filters',
         'Name=instance-state-code,Values=16',
         'Name=tag:env,Values=' + env
     ]
-    describe_instances = json.loads(subprocess.check_output(cmd_parts))
+    return json.loads(subprocess.check_output(cmd_parts))
+
+
+def get_hosts_from_describe_instances(describe_instances):
+    hosts = []
+    for reservation in describe_instances['Reservations']:
+        for instance in reservation['Instances']:
+            hosts.append(
+                Host(public_ip=instance['PublicIpAddress'],
+                     private_ip=instance['PrivateIpAddress']))
+    return hosts
+
+
+def get_inventory_from_file(env):
```

Comment:
Why can't you use the ansible inventory parser?

Reply:
I could have, and I considered it, but ansible inventory doesn't distinguish between a "group" that is really just naming a single host and a group that assigns a role; the inventory parser that comes with ansible would just give me back a bunch of groups and the flat list of hosts (by their `inventory_hostname` (here `host.private_ip`), not by the name of their single-host group (here `host.name`). I found parsing directly from the file that organizes these in a specific way to be easier than working backwards from the flattened group: hosts structure ansible would give me.

I definitely did go down that route first, and I wasn't liking it.

Reply:
Oh and I forgot to say, ansible inventory files as far as i can tell are not actually valid ".ini" files. Python's `ConfigParser` (made for ini files) fails on parsing them.